File: test_data/code/python/1539-kth-missing-positive-number.py
Python
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        i = 1
        miss = 0
        while miss < k:
            if i not in arr:
                miss += 1
                if miss == k:
                    return i
            i += 1
        return None

File: test_data/code/java/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int missingCount = 0;
        int currentNumber = 1;
        
        while (true) {
            if (Arrays.binarySearch(arr, currentNumber) < 0) {
                missingCount++;
                if (missingCount == k) {
                    return currentNumber;
                }
            }
            currentNumber++;
        }
    }
}